src/consensus/consensus.mli

```diff
@@ -0,0 +1,77 @@
+open Crypto
+open Protocol
+
+module Operation_map : Map.S with type key = Protocol.Operation.t
+
+type identity = {
+  secret : Secret.t;
+  key : Key.t;
+  t : Key_hash.t;
+  uri : Uri.t;
+}
+
+type state = private {
+  (* TODO: duplicated on Node.State.t *)
+  identity : identity;
+  trusted_validator_membership_change :
+    Trusted_validators_membership_change.Set.t;
+  pending_operations : float Operation_map.t;
+  block_pool : Block_pool.t;
+  protocol : Protocol.t;
+  snapshots : Snapshots.t;
+}
+
+type t = state
+
+val make :
+  identity:identity ->
+  trusted_validator_membership_change:Trusted_validators_membership_change.Set.t ->
+  t
+
+type effect = private
```

Comment:
why not have a distinct module Effect with type t inside pubic part of the library?

Reply:
To not add another level of indirection, effect is just a property of the consensus, there is no major advantage on adding a module for itself but there is disadvantages, that now to use or handle effects you need to open the Effect module.